idaes/apps/grid_integration/bidder.py

```diff
@@ -1291,3 +1291,395 @@ def _record_bids(self, bids, date, hour, **kwargs):
+            model: bidding model
+
+            date: the date we bid into
+
+            hour: the hour we bid into
+
+        Returns:
+            None
+
+        """
+
+        # record bids
+        self._record_bids(bids, date, hour, Market=market)
+
+        # record the details of bidding model
+        for i in model.SCENARIOS:
+            self.bidding_model_object.record_results(
+                model.fs[i], date=date, hour=hour, Scenario=i, Market=market
+            )
+
+        return
+
+    def _record_bids(self, bids, date, hour, **kwargs):
+        """
+        Record the bis of each time perid.
+
+        Arguments:
+            bids: the obtained bids for this date.
+
+            date: the date we bid into
+
+            hour: the hour we bid into
+
+        Returns:
+            None
+
+        """
+        df_list = []
+        for t in bids:
+            for gen in bids[t]:
+
+                result_dict = {}
+                result_dict["Generator"] = gen
+                result_dict["Date"] = date
+                result_dict["Hour"] = t
+
+                for k, v in kwargs.items():
+                    result_dict[k] = v
+
+                pair_cnt = len(bids[t][gen]["p_cost"])
+
+                for idx, (power, cost) in enumerate(bids[t][gen]["p_cost"]):
+                    result_dict[f"Power {idx} [MW]"] = power
+                    result_dict[f"Cost {idx} [$]"] = cost
+
+                # place holder, in case different len of bids
+                while pair_cnt < self.n_scenario:
+                    result_dict[f"Power {pair_cnt} [MW]"] = None
+                    result_dict[f"Cost {pair_cnt} [$]"] = None
+
+                    pair_cnt += 1
+
+                result_df = pd.DataFrame.from_dict(result_dict, orient="index")
+                df_list.append(result_df.T)
+
+        # save the result to object property
+        # wait to be written when simulation ends
+        self.bids_result_list.append(pd.concat(df_list))
+
+        return
+
+    def write_results(self, path):
+        """
+        This methods writes the saved operation stats into an csv file.
+
+        Arguments:
+            path: the path to write the results.
+
+        Return:
+            None
+        """
+
+        print("")
+        print("Saving bidding results to disk...")
+        pd.concat(self.bids_result_list).to_csv(
+            os.path.join(path, "bidder_detail.csv"), index=False
+        )
+        return
+
+
+class PEMParametrizedBidder(ParametrizedBidder):
+    """
+    Renewable (PV or Wind) + PEM bidder that uses parameterized bid curves.
+    """
+
+    def __init__(
+        self,
+        bidding_model_object,
+        day_ahead_horizon,
+        real_time_horizon,
+        solver,
+        forecaster,
+        renewable_mw,
+        pem_marginal_cost,
+        pem_mw,
+        real_time_bidding_only=False,
+    ):
+        """
+        Arguments:
+            renewable_mw: maximum renewable energy system capacity
```

Comment:
Is this doc string missing arguments (such as bidding_model_object, day_aheady_horizon, ...)?

Reply:
Those are inherited from the 'ParameterizedBidder'. I have added the doc strings for them now.